airflow/www/views.py

```diff
@@ -2012,9 +2012,20 @@ def varimport(self):
         except Exception as e:
             flash("Missing file or syntax error: {}.".format(e))
         else:
+            suc_count = fail_count = 0
             for k, v in d.items():
-                models.Variable.set(k, v, serialize_json=isinstance(v, dict))
-            flash("{} variable(s) successfully updated.".format(len(d)))
+                try:
+                    models.Variable.set(k, v, serialize_json=isinstance(v, dict))
+                except Exception as e:
+                    logging.warning(e)
+                    fail_count += 1
+                else:
+                    suc_count += 1
+            flash("{} variable(s) successfully updated.".format(suc_count), 'info')
```

Comment:
do we want to know which key, value got successfully / unsuccessfully updated instead of the count?

Reply:
As far as I can tell, if an exception is thrown because of Unicode decoding error, it will throw another  exception when trying to log or `flash` the Variable key to the user, which is why I don't think it's possible to show which ones failed to import.